packages/core/src/test/notifications/rendering.test.ts

```diff
@@ -0,0 +1,215 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as sinon from 'sinon'
+import assert from 'assert'
+import { ToolkitNotification } from '../../notifications/types'
+import { panelNode } from './controller.test'
+import { getTestWindow } from '../shared/vscode/window'
+import * as VsCodeUtils from '../../shared/utilities/vsCodeUtils'
+import { readonlyDocument } from '../../shared/utilities/textDocumentUtilities'
+
+describe('Notifications Rendering', function () {
+    let sandbox: sinon.SinonSandbox
+    //const panelNode: NotificationsNode = testNotificationsNode
+
+    beforeEach(function () {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    // util to test txt pop-up under different senarios
+    async function verifyTxtNotification(notification: ToolkitNotification) {
+        const expectedContent = notification.uiRenderInstructions.content['en-US'].description
+        const readonlyDocumentShowStub = sandbox.stub(readonlyDocument, 'show').resolves()
```

Comment:
vscode is showing a document. we can just check that actual fact. we don't need stubs and mocks for that.

`assertTextEditorContains` may be useful, for example. https://github.com/aws/aws-toolkit-vscode/blob/d4d1ae0d065edc4ac795de32996110efb7dbf0dc/packages/core/src/test/testUtil.ts#L381